### PR TITLE
add unit test for function FeatureGateSetFromMap and FeatureGateString

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for function FeatureGateSetFromMap and FeatureGateString
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go [hunk 1]
```diff
@@ -189,3 +189,127 @@ func TestFeatureGateFlagDefaults(t *testing.T) {
...
+			if test.setmapError != "" {
+				if !strings.Contains(err.Error(), test.setmapError) {
+					t.Errorf("%d: SetFromMap(%#v) Expected %v, Got %v", i, test.setmap, test.setmapError, err)
+				}
+			} else if err != nil {
+				t.Errorf("%d: SetFromMap(%#v) Expected nil, Got %v", i, test.setmap, err)
+			}
+			for k, v := range test.expect {
+				if actual := f.Enabled(k); actual != v {
+					t.Errorf("%d:setmap:%#v expected %s=%v, Got %v", i, test.setmap, k, v, actual)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go]: nit: the error message are not uniform, sometimes capital, sometimes not.

**Comment by sttts** [comment]
looks good. Just a small nit.

File: staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go [hunk 2]
```diff
@@ -189,3 +189,132 @@ func TestFeatureGateFlagDefaults(t *testing.T) {
...
+		},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("test String= %s", test.expect), func(t *testing.T) {
+			f := NewFeatureGate()
+			f.Add(featuremap)
+			f.SetFromMap(test.setmap)
+			result := f.String()
+			if result != test.expect {
+				t.Errorf("%d:Test String:%#v Expected %s, Got %s", i, test.setmap, test.expect, result)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go]: can you adapt this test as well to the string patterns in the test above?

File: staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go [hunk 3]
```diff
@@ -189,3 +189,132 @@ func TestFeatureGateFlagDefaults(t *testing.T) {
...
+				testAlphaGate: false,
+				testBetaGate:  false,
+			},
+			setmapError: "unrecognized key:",
+		},
+	}
+	for i, test := range tests {
+		pairs := []string{}
+		for k, v := range test.setmap {
+			pairs = append(pairs, fmt.Sprintf("%s=%t", k, v))
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go]: maps are not ordered. So we get potentially changing test names.

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go]: I would give the tests a better, descriptive name instead.

Comment (hanxiaoshuai) [reply staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go]: @sttts Done Thanks

**Comment by hanxiaoshuai** [comment]
cc @sttts

**Pull request merged**